### ISSUE TITLE
Importing rclone.conf causes app to keep crashing

### ISSUE THREAD

**Issue by leowankerddd** [chenxiaolong/RSAF#65]
Hi,

I imported the rclone.conf that I use on Windows AND with Roundsync. I did this import on both an Android Tablet from Samsung as well as a cell phone from Samsung. Both apps crash when the file is imported. Tab is an S7 and phone is S23. If I manually create a single remote to my Google Drive the app works fine.

I must delete both cache and data from the app to get it to open again. I will try adding a few more remotes manually and see how it goes. Google drive by itself works.

The rclone version is identical as on Windows. Version 1.67.0

The same rclone.conf successfully imports in to Roundsync and all the remotes work.

Something is tripping up the app, just don't know what.

Let me know if there is any further information I need to provide.

Thanks,

Leo.

**Comment by chenxiaolong** [comment]
Hmm, can you upload the `crash.log` file from `/sdcard/Android/data/com.chiller3.rsaf/`? There should not be any sensitive information in there, but please double check to be sure.

**Comment by leowankerddd** [comment]
I plugged my phone in to my computer, navigated to that folder and it was empty.

Here is a picture of the error on the screen, and a 1 minute video of the app crashing.


![Screenshot_20240902_134311_Device care](https://github.com/user-attachments/assets/d54e5905-b24b-4478-8a82-a7069e6351ba)


[https://u.pcloud.link/publink/show?code=XZySKO0Zap7rMHD8QbYrlG2n8Uvkk0n3Hfty](url)

**Comment by chenxiaolong** [comment]
That's interesting. The crash logging is one of the very first things RSAF does during startup. If the log is missing, it must be crashing even earlier than that.

If you're familiar with `adb`, could you grab a logcat immediately after it crashes with:
```bash
adb logcat -d > logcat.txt
```
and upload `logcat.txt`?

Also, could you upload your `rclone.conf` with the usernames and passwords inside replaced with some placeholder value?

**Comment by leowankerddd** [comment]
> f you're familiar with `adb`, could you grab a logcat immediately after it crashes with:
> 
> ```shell
> adb logcat -d > logcat.txt
> ```
> 
> and upload `logcat.txt`?

Yes, I'm familiar with ADB.

Here is the output file you requested. **I hope it helps.**

[logcat.zip](https://github.com/user-attachments/files/16840814/logcat.zip)


[rclone.conf.txt](https://github.com/user-attachments/files/16840867/rclone.conf.txt)

**Comment by chenxiaolong** [comment]
Thanks! This is such a dumb issue on my part:
```
09-02 07:51:56.064 12983 12983 E AndroidRuntime: java.lang.IllegalArgumentException: Max number of dynamic shortcuts exceeded
```
It's crashing due to attempting to create too many Android dynamic shortcuts (the shortcuts that appear when long pressing RSAF's icon on the home screen). RSAF tries to create one for each of your 34 remotes and Android's limit on most devices is only 4.

I'll update RSAF to only create dynamic shortcuts if the user explicitly enables it for a remote. Easier to deal with the limit that way.

**Comment by leowankerddd** [comment]
**Thank you! Much appreciated.**

I'm glad I was able to send you what you needed and it helped you solve the issue.

👍

**Comment by chenxiaolong** [comment]
This is fixed in #66. I'll release version 1.26 shortly, which will include that change.
